### ISSUE TITLE
yolo8-n FPS difference yaml vs .pt

### ISSUE THREAD

**Issue by pie-leroy** [ultralytics/ultralytics#12165]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have a very significant difference in FPS when I do my test and convert my model into a torch : 

```
    model = YOLO("/tmp/yolov8n-seg.yaml") #local file, same as you
    measure_fps(model, "yolo seg 112 yaml", torch.randn(1,3,640,640))
# 60 fps

    model = YOLO("yolov8n-seg.pt") #your weights
    measure_fps(model, "yolo seg 112 ", torch.randn(1,3,640,640))
#120 fps
```

Could you explain why? Is it related to quantization or some other process?

Moreover, when I modify the parameters of yolov8n-seg.yaml by decreasing the number of weights, filters, GFLOPs decreases but my FPS remains almost unchanged,

thanks in advance!

**Comment by glenn-jocher** [comment]
@pie-leroy hello!

The FPS discrepancy you're observing between the model loaded from a YAML file and the pre-trained `.pt` model might stem from several factors. When using the YAML file, the YOLO model needs to create and initialize the network based on the architecture specifications in the YAML. This process might not include certain optimizations that are already incorporated into the pre-trained `.pt` models, such as optimized layer operations or pre-compiled structures which can significantly boost the inference speed.

In terms of your modifications in the YAML file not affecting the FPS, it's possible that changes in parameters like numbers of weights or filters don't directly translate into FPS gains due to other bottleneck processes during inference, such as I/O operations or other non-computational overheads.

If your primary focus is on maximizing FPS, I'd recommend experimenting with different export formats (like ONNX or TensorRT) that may provide further optimizations specifically designed for speed.

Hope this clarifies your questions! 🙂

**Comment by pie-leroy** [comment]
Thank you for these explanations.

I'm actually trying to work from the backbone of the classification model by adding a specific head. 

I was therefore surprised by the small difference between my scratch-trained yolov8 classification model and your optimised segmentation model.

Is there any documentation on all the optimisations made? 

At the end of my process I use tensorrt for inference.

Thanks for your help! :)

**Comment by glenn-jocher** [comment]
Hello @pie-leroy,

Great to hear about your work on adapting the YOLOv8 classification backbone with a custom head!

Regarding the optimizations: Our pre-trained models often include trained optimizations tailored for performance, which may not be directly visible in the architecture file. These optimizations can include things like pre-tuned batch normalization layers, advanced training techniques, and more.

We don't have detailed public documentation on all specific optimizations, as many are intrinsic to how the models are trained and prepared.

Since you're using TensorRT for inference, you're already on a good path for maximizing performance! Make sure your model is fully compatible with TensorRT, and consider employing techniques such as precision calibration for further optimization.

Keep up the great work, and don't hesitate to reach out if you have more questions! 😊
